### PR TITLE
ERT Timelocks and Tweaks

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Each ERT role, except for the standard squaddie one, now has a 5 hour timelock based on its marine equivalent. For instance, a UPP Minigunner needs 5 hours of SG hours, a CLF SL needs 5 marine SL hours, etc.

Additionally, each prob chance to get a role based on how many heavies there are, like engineers, SGs, and Specialists splitting a slot, have been split into different vars so each distress call has generally the same layout.

Split the UPP Minigunner and UPP Shotgun Specialist into two different equipment presets, for ease of spawning.

## Why It's Good For The Game

Having a complete baldie as a medic, engineer, or worse, squad leader, just isn't fun. This doesn't eliminate the chance of that, but it drastically reduces it. It also doesn't exclude baldies from joining ERTs, they just get the standard squaddie roles instead.

## Changelog

:cl: Geeves
add: Special ERT roles now have (5 hour) timelocks, the standard squaddie role for each ERT doesn't, so no one is prohibited from joining, just from getting special roles.
add: Where ERTs previously had a probability to get either two SGs, one SG one spec, two specs, they will now always get one SG, one spec.
admin: Split the UPP Minigunner and UPP Shotgun Specialist into two different equipment roles.
fix: Fixed up some of the UPP backstory text.
/:cl:

### CODE REVIEW

**Comment by PurpleCIoud** [comment]
This is a good way to prevent that

**Comment by stanalbatross** [comment]
15 hours is far, far too much (it takes about 15 minutes to learn how to play SG).

Also, being put into an ERT role you don't know is a fun and new experience - back in 2020, getting medic in a cryo ERT inspired me to get out of my comfort zone (PFC) and actually learn how to play medical.

I would say reduce these locks by a LOT (I don't even think I have 15 hours on SG)

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
see what stan said above

plus, timelocks do not prevent incompetence. If the issue is in admin events, consider adding an option for admins when summoning an ERT to be able to customize the timelock.

**Comment by Watermelon914** [comment]
I don't think I agree with it then, we should discuss this properly

**Comment by tempelers** [comment]
This is a good change. I'm tired of the random ass PFC getting PMC SL and then the PMC medic has to win the game.

**Comment by tempelers** [comment]
MARSOC should also receive a timelock within the code.

I'm tired of seeing baldsoc cade hugging with the Marines. It just gets very grading on my soul.

I would do a 20-50 hour timelock from general Marines (aka any squad marine role).

**Comment by MusselMain** [comment]
I think 8 or 10 hours makes the most sense, 15 is a bit intense (I do have the hours on all roles for that, but I think it can be problematic for some of the newer players), I do think this update is a good change though, it can help lower the amount of rounds I see where our medic goes LOOC: "help first time medic" which I see 7 times out of 10, or the SG is some PVT level player who doesn't even know how to wield their weapon, or the classic SL who doesn't realize they have orders and who immediatly fucks off from their squad and dies, which is a issue as ERT's lack SL trackers.

**Comment by MusselMain** [comment]
> 15 hours is far, far too much (it takes about 15 minutes to learn how to play SG).
> 
> Also, being put into an ERT role you don't know is a fun and new experience - back in 2020, getting medic in a cryo ERT inspired me to get out of my comfort zone (PFC) and actually learn how to play medical.
> 
> I would say reduce these locks by a LOT (I don't even think I have 15 hours on SG)

They have, I think in the case of ERT's that are called in via shuttle, we absolutely need this I've seen far too many newer players get ahold of a ERT role they're new to and who promptly throw the fun for the rest of the ERT as they either refuse to revive anyone (because they have no idea how medic works), or rush the queen with a unwielded SG/UPP minigun, resulting in their immediate death.

**Comment by tempelers** [comment]
> > MARSOC should also receive a timelock within the code.
> > I'm tired of seeing baldsoc cade hugging with the Marines. It just gets very grading on my soul.
> > I would do a 20-50 hour timelock from general Marines (aka all squad marine role).
> > Alternatively, you could maybe make it so only people have played MARSOC before can get randomly chosen.
> 
> I like the first part of your idea, but the second is a terrible idea, how would veterans who've never been MARSOC before get a chance to roll MARSOC? It just seems like a Ghetto Whitelist with extra steps.

Admins can spawn you as MARSOC, but I see your complaints.

**Comment by MusselMain** [comment]
> I don't think any roles should require specialist or smartgunner times as there's a few issues with that
> 
> * Not everyone gets to play specialist or smartgunner often, it's often taken before anyone can get it (and thus can't get many hours in it)
> * Marine specialist or marine smartgunner is not always everyone's thing, but they're entirely different things from their ERT counterparts.
> * Many of these roles are just easy point and shoot things. WY deathsquad smartgunner is the same as any other deathsquad member just with a better/different gun, same with PMC sniper and same with UPP minigunner

I have to disagree, while the UPP Minigunner does have a lower skill floor, SG's and PMC Snipers have many traits that are different from "point and shoot", SG's have a whole power usage system, NVG, farsight, APvsStandard ammo, and a whole lot of other traits. I've seen countless times where a PMC SG tries to look in their backpack and accidentally removes it, resulting in their Eyepiece falling off and they then they forget it exists, (especially since it's not as noticeable if you're a newer player). Whereas PMC Snipers have a both a NVG eyepiece, and a sniper rifle that acts very different from the standard Sniper Specialist. Once again, I can tell new players are playing PMC Spec, when they don't use it's wall penetration to kill xenos (yes it can penetrate walls and keep going), or try and use it as a budget Type 23 riot shotgun at close range, resulting in their death as well. 

Long story short all of these timelocks make sense, Medics need to know how to use their gear, SL's need to know how to at minimum issue orders and command a Squad, SG's need to know how to use their Smart gun's many different modes, and specialists need to understand their rifle's abilities, and range of application. 

If you want more SG/Spec hours, simply play on low pop, every day(/night) for about 7 hours, there's a average of 40-70 players online, and I consistently get SG and Spec when set to high. and in the case of medics, I suggest going nurse and shadowing a Doctor until you know the basics of frontline medicine, then going Medic to get your hours for ERT Medics.

**Comment by tempelers** [comment]
I think test merging this PR after the VC TM ends could be helpful to test out the changes.

**Comment by neroisvcute** [comment]
> simply play on low pop, every day(/night) for about 7 hours

Most people with lives can't play everyday for about seven hours. Most people don't even play every day.

Why is this even an issue? It's ERT, that means its the end of the round so who cares. Let some baldo play with an SG or sniper rifle. I could see making medics have hours but with the new nurse timelock nobody new even wants to touch medical. I swear in a year there will be a timelock for MTs. Timelocks do nothing, ERTS are going to rush off and die everytime. Even robust medics will unga as ERT because who wants to heal on ERT.

Timelocks solve nothing other than fucking over new players. Bad PR.

**Comment by neroisvcute** [comment]
Also MARSOC mains in here are annoying and cringe. MARSOC means the round is over. Nobody cares if you get six kills as MARSOC or sit behind cades with marines. Why do we want to make new players grind for something that lasts five minutes. Let them run around in OP gear. People are so self centered on this server now.

"I saw a bald noob WASTE time as MARSOC for a WHOLE FIVE MINUTES. I COULD'VE KILLED SIX XENOS AND HAD SO MUCH FUN." Seriously that shit is cringe, its a game. Let new players have fun, fuck timelocks. Esp for ERTS which only ever happen when the round has less than ten minutes.

Don't like watching a new player play for at most fifteen minutes? Do something else.

**Comment by MusselMain** [comment]
> > simply play on low pop, every day(/night) for about 7 hours
> 
> Most people with lives can't play everyday for about seven hours. Most people don't even play every day.
> 
> Why is this even an issue? It's ERT, that means its the end of the round so who cares. Let some baldo play with an SG or sniper rifle. I could see making medics have hours but with the new nurse timelock nobody new even wants to touch medical. I swear in a year there will be a timelock for MTs. Timelocks do nothing, ERTS are going to rush off and die everytime. Even robust medics will unga as ERT because who wants to heal on ERT.
> 
> Timelocks solve nothing other than fucking over new players. Bad PR.

I have to heavily disagree, this is a server where players learn by being taught, we have SEA's for a reason, and MANY new players are playing SS13 for the first time on CM because it's the top of the list, this is why it makes sense to ease them into the server instead of letting them sink or swim and suffer the consequences, this is exactly why timelocks exist to force newer players into a role where they won't be getting screamed at by half the server for choosing a role that holds a lot of responsibility, also I could be wrong as I've been playing for three years and have all the timelocked roles opened, but last I heard Nurses weren't timelocked, If they are it's probably for a valid reason, if you're talking about introducing new players to the craziness of CM then giving them a role where they sit around and clean floors/fix walls (MT), or heal people/learn medicine (Nurse) doesn't give a proper justice, this also allows SEA's to find and assist new players faster, as they not only have a PVT ranking, but also are usually walking around naked near cryo (and yes I know timelocks suck for people who can't spare 3 hours every day for CM, I am one of those people, It took me 6 months to unlock HALF the roles, so I get it, but it's part of the process for litterally every server). I do agree with you that standard ERT foot-soldiers should be freely accessible for anyone (even 0 hour privates), however I have to disagree with you saying that Medics, Snipers, and SG's should be allowed to play at any level, I've seen first hand players who get a role like SG, Sniper, or UPP heavy and are so new they don't understand how to wield their gun (let alone reload it), I always feel bad for these players as not only do some other players lose their shit when they die, but also they could've learnt significantly more by playing Private, where they will actually learn the mechanics of the game, and work their way up. 

I also have to somewhat disagree with your statement about the round being "pretty much over" after the xenos board the Almayer, I've been on this server for 3 years, and I've seen hundreds of boardings, and I can guarantee you that after the xenos board a round can last anywhere from 10 minutes to 2 HOURS longer (I've seen this twice), and I can also promise you that a good ERT can make a massive difference, Hell, last month I saw a freelance squad of 5 kill 14 xenos WITHOUT SUPPORT (as they all had at least a couple hours of experience, and their medic was reviving men at a insane rate), before succumbing to the final 4 xenos (it was a VERY close round), and they're the second weakest ERT (I'm not counting XRT's or Pizza Delivery), I understand that giving new players timelocks might seem like it's going to result in players opening their preferences and quitting at the sight of the time requirements for each role. but in most of the cases I've seen (I'm talking about the PVT's I've taught the basics who always ask why that one guy has a big LMG), they instead see the effectiveness of all these cool weapons they don't have access to yet (specialists and SG weapons), and immediately get more invested in trying to get the time requirements for those roles, which literally ends up causing new players to unintentionally learn through trial and error and figure out more about specialists and SG's by simply fighting beside them for a few hours. 

I do want to clarify that I'm someone who Isn't a SEA, but I try to teach every single private I see (unless they're a 4 and a half-hour private who clearly knows where the food, gear, and weapons are), because I hate seeing a new player who's been walking around cryo for half a hour and clearly doesn't know where to go, who then quit the game because they didn't get the help they needed, and I feel that while timelocks may be harsh they're put in place for a reason.

**Comment by MusselMain** [comment]
> Also MARSOC mains in here are annoying and cringe. MARSOC means the round is over. Nobody cares if you get six kills as MARSOC or sit behind cades with marines. Why do we want to make new players grind for something that lasts five minutes. Let them run around in OP gear. People are so self centered on this server now.
> 
> "I saw a bald noob WASTE time as MARSOC for a WHOLE FIVE MINUTES. I COULD'VE KILLED SIX XENOS AND HAD SO MUCH FUN." Seriously that shit is cringe, its a game. Let new players have fun, fuck timelocks. Esp for ERTS which only ever happen when the round has less than ten minutes.
> 
> Don't like watching a new player play for at most fifteen minutes? Do something else.

Oh boy, I was trying to act cool with your last section, but this seems a bit aggressive, please calm down and think before you type next time, were here to discuss these changes, not fight to the death over them. 

MARSOC is usually called in by either Admins for a Event, or by admins who want to rap up a round that would've otherwise dragged on, they are NOT a ERT that can be easily called in by anybody in CIC who knows how (and who has the required access), and during certain events I've seen them active for up to a hour and 10 minutes. I don't at all think that MARSOC should be whitelisted or massively time restricted, however I do think it should have a 10 hour timelock, meaning you need a collective total of 10 hours between any of the marine combat role (this includes: SL, RTO, SG, SPEC, Medic, Engineer, and Rifleman). This is not a high bar, for reference moving from PVT to PFC only takes 5 hours, so basically double that and you can now play MARSOC. Even if you only have a hour to spare every day, that's less then two weeks, and is in place to ensure MARSOC players aren't so new that they're going to get insulted and degraded by fellow players, and so that they at least have enough time invested in the game to know how to use their stimulants and rifle, while not accidentally switching to aim mode, locking onto a SEA or CO's head, shooting them to death and getting banned (which has happened before, as it's often misinterpreted as griefing instead of lack of knowledge).

While I do agree, the end of the round can get crazy, it's not a hands down Free-For-All, and the server does have rules, shooting fellow players often enough will still results in a EORG ban, and most players don't give up simply because they think they're losing.

I have to mention, you say this change will hurt newer players, and that none of us care about fellow Baldies who are new to CM, and while you are right about some players (I'm thinking of two of them by name right now) being complete dickheads who lose their mind and PB medics when they prioritize a dead marine over their slight injury (yes I did see this happen, and I reported the dickbag), the vast majority are players who will rush a choke by themselves to drag out a dead PVT for revival, and we have timelocks to prevent new players from being assigned a role where they will have no idea what they are doing, and will feel severe stress because they're expected to preform the role to a somewhat competent degree without the knowhow or experience to do so. 
This is why we have different progressions for different sections, Medical: Nurse (plenty of time to learn, no demands from other players), Doctor/Medic (medium demands, less room for error), CMO (medium to high demands, very little room for error, In charge of all most of medical's decisions). Engineering: MT (no demands, plenty of time to learn), OT/Marine Engineer (Medium demands, little room for error), CE (low to high demands, slight room for error, In charge of part of Engineering's decisions). Piloting: DCC (No demands/ low demands, plenty of room to learn, low risk of error or severe FF), DP (Medium to high demands, a decent amount of room to learn, high risk of error/high risk of FF). 

I could go on about command and marine sections as well, but I'm sure you get the gist, every role aside from CL (it's not really needed, but RP'ing does take some creativity), and VC (there's no Junior/cadet VC, so new players have to lean on a somewhat knowledgeable VC to teach them the ropes) have a entry level role to ease new players into properly learning the skills related to that section, and I think that ERT roles should follow a similar path. My idea is that it look something like this: CLF/Freelance/UPP/PMC standards: 0 hours (no timelock), UPP Heavy's/PMC Specialists/PMC Smart Gunners: 5 hours (collectively of any Marine combat role), UPP/PMC/CLF/Freelance medic: 10 hours (collectively of any medical role), CLF Cell leader/PMC Leader/Freelance Warlord/UPP Master Serzhant: 10 hours (Collectively of any Marine combat role). 

This will both allow new players to quickly acquire these ranks (as they're relatively low timelocks), while also preventing new players from being thrown into a role they know nothing about and suffering major backlash and stress from having their whole squad blaming them for the demise of their squad (which is something that almost never happens for standard Riflemen in ERT's with the exception of grenade spamming Freelancers).

**Comment by stanalbatross** [comment]
conflicts

**Comment by tempelers** [comment]
> Also MARSOC mains in here are annoying and cringe. MARSOC means the round is over. Nobody cares if you get six kills as MARSOC or sit behind cades with marines. Why do we want to make new players grind for something that lasts five minutes. Let them run around in OP gear. People are so self centered on this server now.
> 
> "I saw a bald noob WASTE time as MARSOC for a WHOLE FIVE MINUTES. I COULD'VE KILLED SIX XENOS AND HAD SO MUCH FUN." Seriously that shit is cringe, its a game. Let new players have fun, fuck timelocks. Esp for ERTS which only ever happen when the round has less than ten minutes.
> 
> Don't like watching a new player play for at most fifteen minutes? Do something else.

MARSOC is a major event. It's not something you can get randomly. 9/10 times MARSOC is spawned in the middle of the round to do an event or get an objective done.

Whiteout is the Admin ERT that can only really be spawned at the end of the round. I can understand your concerns with ERTs being for new players to get interested in roles, but the whole point of MARSOC is that you're a highly trained marine man/woman. If I die to a runner that doesn't make sense lore wise.

MARSOC is an especially rare ERT, so having it wasted when they can wipe out the hive is disheartening. 

Plus, your tone is especially irritating. We're all friends here, there's no reason to call people annoying or cringe.

**Comment by Nanu308** [comment]
5h is fine

**Comment by stanalbatross** [comment]
code conflicts

**Review by stanalbatross — APPROVED** [review]
